src/main/kotlin/network/rs485/logisticspipes/config/ServerConfiguration.kt

```diff
@@ -44,29 +44,29 @@ import java.lang.reflect.Type
 import java.util.*
 import kotlin.collections.HashMap
 
-private class ServerConfigurationAdapter : JsonSerializer<Map<PlayerIdentifier, ClientConfiguration>>, JsonDeserializer<Map<PlayerIdentifier, ClientConfiguration>> {
-    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Map<PlayerIdentifier, ClientConfiguration>? {
+private class ServerConfigurationAdapter : JsonSerializer<Map<PlayerIdentifier, PlayerConfiguration>>, JsonDeserializer<Map<PlayerIdentifier, PlayerConfiguration>> {
+    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Map<PlayerIdentifier, PlayerConfiguration>? {
         if (json.isJsonNull) {
             return null
         } else if (!json.isJsonObject) {
             throw JsonParseException("Expected an object from a map")
         }
 
-        return HashMap<PlayerIdentifier, ClientConfiguration>().run {
+        return HashMap<PlayerIdentifier, PlayerConfiguration>().run {
```

Comment:
Why not replace the whole return block with something like

```kotlin
return json.asJsonObject.entrySet().associate {
  if (!it.value is JsonObject) throw JsonParseException("Expected values in map object to be objects")
  val k = PlayerIdentifier.get(obj["name"].asString, UUID.fromString(it.key))
  val v = context.deserialize(it.value["config"], PlayerConfiguration::class.java)
  k to v
}
```

Not sure if the smart cast to JsonObject will work here or if you still need to do asJsonObject, but something along these lines should be way less convoluted imo. (Untested code, wrote this in my text editor)

Reply:
btw. for refactorings: just grab the code and do it :) This change doesn't need to be in this PR, because this PR barely touched the code.

Reply:
Also, instead of
```kotlin
T.run {
  // …
  this
}
```
use
```kotlin
T.apply {
  // …
}
next time, it does the same and you don't have to return this